### DISCUSSION TITLE
Setting value with initial values does not update selected tags when adding or removing items.

### DISCUSSION THREAD

**Discussion by alveshelio** [JedWatson/react-select#4617] [Q&A] [answered]
Hi,

I'm having a hard time understanding how to set initial values. I've read a couple of issues and they all pointed to the same solution, using `value` to set the initial values alongside with the onChange. I don't know how I should use the onChange and if I only set `value` to the initial list of addresses, then whenever I want to add or remove items the tags do not get updated, even though the selected items get updated.

I've created this code sandbox: https://codesandbox.io/s/snowy-https-93dd1?file=/src/SearchAddressProvider.tsx

And if you look in file `MultiSelectAddress.tsx` I'm passing an array and if you search for an address and add it, the list of tags does not get updated but if I console the selected items they are added.

I'd be very grateful if someone who faced this issue and was able to fix it could either update the code sandbox with a working example or tell me how I could fix this issue.

Thank you very much for your help

**Comment by Rall3n** [comment, marked as answer]
Hello @alveshelio,

I hope you don't mind me converting your issue to a discussion. It seemed more fitting as your query does not include an issue in the library but instead a question regarding implementation.

Your CodeSandbox example seems to be unavailable, but I had a look before and can probably offer a solution.

You used a prop called `initialSelectedAddresses` in your `MultiSelectAddress` component. This prop should contain your inital value of the `Select` component.
In the `MultiSelectAddress` component you also have a state called `selectedAddresses`. This seems to be your state to which you want the initial values to be applied.

As I could see (and remember) you used your `initialSelectedAddresses` as the value of the `Select` component, but updated the `selectedAddresses` state in the `onChange` handler. This would explain why the `Select` component never shows the new values.

I recommend using the `selectedAddresses` state as the value of the `value` prop on the `Select` component.

**Reply by alveshelio** [reply]
Hi @Rall3n,

Thank you very much for taking the time to answer. Next time I'll make sure to open a discussion instead an issue :)
